### ISSUE TITLE
Regression since 2.6.3.Final in OpenShift extension

### ISSUE THREAD

**Issue by edeandrea** [quarkusio/quarkus#23417]
### Describe the bug

Seems when the openshift extension generates the resources its no longer generating the `ImageStream` resource properly.

See https://github.com/quarkusio/quarkus-super-heroes/blob/5a805fadaeff91572345777775a943d5324c67be/rest-fights/deploy/k8s/native-java17-openshift.yml#L83-L91 (generated with the openshift extension on Quarkus `2.7.0.Final`) vs https://github.com/quarkusio/quarkus-super-heroes/blob/c2e744ec28defd9d97b4ba578d1e0f16eeb3ea8d/rest-fights/deploy/k8s/native-java17-openshift.yml#L83-L103 (generated with the openshift extension on Quarkus `2.6.3.Final`).

### Expected behavior

The generated `ImageStream` resource doesn't have a `metadata.name` value. Even still as you can see from the 2 links

**Comment by geoand** [comment]
@iocanel has this already been fixed?

**Comment by iocanel** [comment]
> @iocanel has this already been fixed?

What we fixed was adding the missing `ImageStream`. In this particular case, the `ImageStream` is there but is missing the `metadata.name`.

**Comment by edeandrea** [comment]
It doesn't seem though like the labels on it are correct. The `name: fights-db` under `labels` isn't correct either (both in the 2.6.3 & 2.7.0 version. I'm not sure where it is pulling that label from? Shouldn't it be `name: rest-fights`?

**Comment by iocanel** [comment]
The missing `metadata.name` should be fixed by: https://github.com/quarkusio/quarkus/commit/ffc5957cd863c44f8e6a9cf94b415011ac109c37#diff-2f7f8b1e6f4a44f6f156a44d9a60ec9f4a233e7874e8059fb0c627771a505fc4R271

**Comment by iocanel** [comment]
> It doesn't seem though like the labels on it are correct. The `name: fights-db` under `metadata.labels` isn't correct either (both in the 2.6.3 & 2.7.0 version). I'm not sure where it is pulling that label from? Shouldn't it be `name: rest-fights`?

I can't tell about the labels, unless I see the configuration. Let me have a look at your project....

**Comment by edeandrea** [comment]
> I can't tell about the labels, unless I see the configuration. Let me have a look at your project....

The `application.properties` file is at https://github.com/quarkusio/quarkus-super-heroes/blob/main/rest-fights/src/main/resources/application.properties and the command used to generate would be

```shell
rest-fights/mvnw -f rest-fights/pom.xml versions:set clean package -DskipTests -DnewVersion=native-java17-latest -Dquarkus.container-image.tag=native-java17-latest -Dquarkus.kubernetes.deployment-target=openshift -Dquarkus.kubernetes.version=native-java17-latest -Dquarkus.kubernetes.ingress.expose=true -Dquarkus.kubernetes.resources.limits.memory=128Mi -Dquarkus.kubernetes.resources.requests.memory=32Mi '-Dquarkus.kubernetes.annotations."app.quarkus.io/vcs-url"=https://github.com/quarkusio/quarkus-super-heroes' '-Dquarkus.kubernetes.annotations."app.quarkus.io/vcs-ref"=main' -Dquarkus.openshift.version=native-java17-latest -Dquarkus.openshift.route.expose=true -Dquarkus.openshift.resources.limits.memory=128Mi -Dquarkus.openshift.resources.requests.memory=32Mi '-Dquarkus.openshift.annotations."app.openshift.io/vcs-url"=https://github.com/quarkusio/quarkus-super-heroes' '-Dquarkus.openshift.annotations."app.openshift.io/vcs-ref"=main' -Dquarkus.knative.version=native-java17-latest '-Dquarkus.knative.labels."app.openshift.io/runtime"=quarkus' -Dquarkus.knative.resources.limits.memory=128Mi -Dquarkus.knative.resources.requests.memory=32Mi '-Dquarkus.knative.annotations."app.openshift.io/vcs-url"=https://github.com/quarkusio/quarkus-super-heroes' '-Dquarkus.knative.annotations."app.openshift.io/vcs-ref"=main'
```

https://github.com/quarkusio/quarkus-super-heroes/runs/5054787642?check_suite_focus=true#step:4:11282

**Comment by iocanel** [comment]
The labels thing must be a bug triggered by the fact that there are multiple `Deployment` resources in the final manifest. This is something that we should do a better job at handling. I'll work on it ASAP. 

The missing `metadata.name` should be fixed in the main branch already. Let me checkout your project to verify.

**Comment by edeandrea** [comment]
> The labels thing must be a bug triggered by the fact that there are multiple Deployment resources in the final manifest. This is something that we should do a better job at handling. I'll work on it ASAP.

Yeah if in `src/main/kubernetes/openshift.yml` it seems that whatever labels/annotations are on the last `Deployment` in there are carried forward. They shouldn't be.

I would think they should only be carried forward if the `Deployment.metadata.name` (or `DeploymentConfig.metadata.name`) matches the name of any of the generated resources.

**Comment by iocanel** [comment]
The missing `metadata.name` is already fixed on master. And the labels are indeed broken, as it seems to pickup the labels from your postgres deployment, instead of the app.

**Comment by edeandrea** [comment]
> The missing `metadata.name` is already fixed on master.

Awesome! Will that make it into 2.7.1?

**Comment by gsmet** [comment]
@iocanel what is the PR fixing this `metadata.name` issue?

**Comment by iocanel** [comment]
@gsmet: It's #23239

**Comment by gsmet** [comment]
OK, so I can confirm this particular fix is planned for 2.7.1.Final.

**Comment by edeandrea** [comment]
> And the labels are indeed broken, as it seems to pickup the labels from your postgres deployment, instead of the app.

Should I open a new issue @iocanel for this?
